Engine/src/Razix/Core/OS/RZInput.h

```diff
@@ -11,10 +11,16 @@ namespace Razix {
     class RAZIX_API RZInput
     {
     public:
-        // TODO: Use a better API convention for selecting input API implementation
-        // TODO: ReWrite this using a simple static function design pattern rather than using Impls, this is cumbersome to extend
-        /* Selects the GLFW Input class for polling and reporting input events */
+        // Platform input manager selection methods
+        // These allow selecting different input implementations at runtime or compile time
         static void SelectGLFWInputManager();
+#ifdef RAZIX_PLATFORM_WINDOWS
+        static void SelectWindowsInputManager();
+#endif
+        // Future platform methods can be added here:
```

Comment:
Remove these s_Instance design, similar to RZFileSystem we can bake these classes during compile time, no need to switch during runtime, remove SelectManager mechanism and have static functions and the per-platform implementation in .cpp class respedctive we choose the backend in premake file razix_engine.lua, default is GLFW and windows will be enabled using premake options and for others we can let premake choose it remove the select API